pkg/arvo/sys/vane/eyre.hoon

```diff
@@ -2103,6 +2117,23 @@
       =^  moves  server-state.ax
         (on-channel-timeout i.t.t.wire)
       [moves http-server-gate]
+    ::
+        %heartbeat
+      =/  on-channel-heartbeat
+        on-channel-heartbeat:by-channel:(per-server-event event-args)
+      =/  handle-response
```

Comment:
You probably want to move most of this logic into `+on-channel-heartbeat` itself. You have access to `+handle-response` there and the logic handling this probably shouldn't be split into two places.